### PR TITLE
feat: add `extent` transform

### PR DESCRIPTION
**Pull request by lsh**
Adds an interface to use Vega's [`extent` transform](https://vega.github.io/vega/docs/transforms/extent/).

### CODE REVIEW

File: site/docs/transform/extent.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  ...
+  "transform": [
+    {"extent": ...} // Fold Transform
+     ...
+  ],
+  ...
+}
+```
+
+## Fold Transform Definition
```
Comment (domoritz) [inline site/docs/transform/extent.md:22]:
```suggestion
## Extent Transform Definition
```

File: src/compile/data/extent.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {ExtentTransform as VgExtentTransform} from 'vega';
+import {ExtentTransform} from '../../transform';
+import {duplicate, hash} from '../../util';
+import {DataFlowNode} from './dataflow';
+
+/**
+ * A class for flatten transform nodes
```
Comment (domoritz) [inline src/compile/data/extent.ts:7]:
```suggestion
 * A class for extent transform nodes
```

File: src/transform.ts [hunk 3]
```diff
@@ -390,6 +390,19 @@ export interface FoldTransform {
...
 
+export interface ExtentTransform {
+  /**
+   * The field of which to get the extent.
+   */
+  extent: FieldName;
+
+  /**
+   * The output parameter produced by the extent transform.
+   * __Default value:__ `"extent"`
```
Comment (domoritz) [inline src/transform.ts:401]: Can there really be a default? Would that mean you pass undefined or null? Maybe we should not make this optional.

**Review by domoritz — APPROVED** [review]
Looks good. I'd like to have a quick check from @kanitw before merging.

File: src/transform.ts (hunk 3 above)
Comment (lsh) [reply src/transform.ts]: I agree. Updated to fix this.

File: site/docs/transform/extent.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+And the transform:
+
+```json
+"transform": [
+  {"extent": "b", "param": "b_extent"}
+]
+```
+
+this example produces the output `param` will look like:
```
Comment (kanitw) [inline site/docs/transform/extent.md:48]: Probably better say the output `b_extent` param will be `[19, 91]`? 

(The following JSON in the block isn't really produced anywhere?)

File: src/compile/data/extent.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    super(parent);
+    this.transform = duplicate(transform);
+  }
+
+  public dependentFields() {
+    return new Set([this.transform.extent]);
+  }
+
+  public producedFields() {
+    return new Set([this.transform.param]);
```
Comment (kanitw) [inline src/compile/data/extent.ts:24]: This isn't a field produced.

**Pull request merged**